src/lib/sidenav/drawer.ts

```diff
@@ -249,7 +252,8 @@ export class MatDrawer implements AfterContentInit, AfterContentChecked, OnDestr
      */
     this._ngZone.runOutsideAngular(() => {
         fromEvent<KeyboardEvent>(this._elementRef.nativeElement, 'keydown').pipe(
-            filter(event => event.keyCode === ESCAPE && !this.disableClose)
+            filter(event => event.keyCode === ESCAPE && !this.disableClose),
+            takeUntil(this._destroyed)
```

Comment:
I'm not sure this is necessary. These events are supposed to live as long as the element is in the DOM. They should be cleaned up automatically when the element is destroyed.

Reply:
Okey you mean instead of manually remove the eventListener a garbage collect is able to remove the eventListener because the source object reference no longer exists, is that correct?

Reply:
```
RxJs / fromEvent:
...
function setupSubscription(sourceObj, eventName, handler, subscriber, options) {
    var unsubscribe;
    if (isEventTarget(sourceObj)) {
        var source_1 = sourceObj;
        sourceObj.addEventListener(eventName, handler, options);
        unsubscribe = function () { 
            return source_1.removeEventListener(eventName, handler, options); 
        };
    }
```
I guess it's a matter of preference. I can see why this is happening because sourceObj is this._elementRef and that reference will be destroyed, even though this would solely rely on gc. If this is preferable I assume someone can close this issue.

Reply:
It's not a big deal to keep it in, I just wanted to explain the reasoning behind it. That being said, before marking it as merge ready, you need to rebase it to fix the CI failure and you also need to add `this._destroyed.complete()` after the `_destroyed.next` call.